[FIX] l10n_ar_ux, l10n_ar_account_withholding: auto compute rate on invoices, new constraint in vendor bills

zaoral opened the pull request:
Ticket 71754

roboadhoc commented:
[Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/odoo-argentina/pull/824).

jjscarafia commented on l10n_ar_ux/models/account_move.py:
```diff
@@ -95,36 +95,26 @@ def _post(self, soft=True):
-                accounting_date = inv.date
-                inv.date = old_date
-                inv.date = accounting_date or today
+        other_currency_ar_invoices.filtered(lambda x: not x.l10n_latam_use_documents)._set_afip_rate()
```
1. revisar que, si está andando, pareciera que el _set_afip_rate no es necesario volver a llamarlo (ya el llamado original debería actualiar cotizacion), mantendríamos el llamado para las sin documentos de abajo
2. si no está funcionando entonces el método debería quedar algo así:
3. 
```
    def _post(self, soft=True):
        """ Estamos sobreescribiendo este método para hacer cosas que en odoo oficial no se puede tanto previo como posterior a la validación de la factura. """
        other_currency_ar_invoices = ar_invoices.filtered(lambda x: x.currency_id != x.company_currency_id and not x.l10n_ar_currency_rate)
        other_currency_ar_invoices.._set_afip_rate()

        res = super()._post(soft=soft)

        return res
```

zaoral committed:
[FIX] l10n_ar_ux: Cotizacion en facturas

Modificamos el parche apra corregir cotizacion de facturas para que solo
aplique a facturas de clientes, porque en facturas de proveedor estamos
teniendo un problema que no blanquea los impuestos fijos (percepciones)
al validar la factura.

Ahora:
- Recalcula cotiz solo para ventas y funciona para todos los coasas
- Salta error de validacion en facturas de proveedor si detectamos
  que las lineas usan una cotizacion distinta a la que se va a informar

Task 71754

jjscarafia commented:
cerramos en favor de https://github.com/ingadhoc/odoo-argentina/pull/843
